lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -734,6 +734,26 @@ def test_problem_grade_report(self):
             ))
         ])
 
+    def test_problem_grade_report_valid_columns_order(self):
+        """
+        Test that in the CSV grade report columns are placed in the proper order
+        """
+        problem_names = [u'Homework 1: Problem - pröblem_a_url']
+        for i in xrange(30):
+            problem_url_a = 'test_problem_a_%d' % i
+            self.define_option_problem(problem_url_a, parent=self.vertical_a)
+            title = 'Homework 1: Problem - %s' % problem_url_a
+            problem_names.append(title)
+        problem_names.append(u'Homework 1: Problem - pröblem_b_url')
+
+        header_row = [u'Student ID', u'Email', u'Username', u'Final Grade']
+        for problem in problem_names:
+            header_row += [problem + ' (Earned)', problem + ' (Possible)']
+
+        with patch('instructor_task.tasks_helper._get_current_task'):
+            upload_problem_grade_report(None, None, self.course.id, None, 'graded')
+        self.assertEquals(self.get_csv_row_with_headers(), header_row)
```

Comment:
@nedbat A regular `dict` does not track the insertion order. If you run this test without my [fix](https://github.com/edx/edx-platform/pull/11941/files#diff-9b7fcb710c553d15f0d138a19ec082d0L839) you will see that it fails on assert line. In case of [2 columns](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor_task/tests/test_tasks_helper.py#L712) of course you will not see any difference. In case of 30 columns test will be failed